src/test/scala/tech/cryptonomic/conseil/tezos/DataTypesTest.scala

```diff
@@ -0,0 +1,202 @@
+        fields = None,
+        predicates = Some(List(Predicate("valid", OperationType.in))),
+        orderBy = None,
+        limit = None,
+        output = None,
+        aggregation = None
+      )
+
+      val result = query.validate("test", tpdo)
+
+      result.futureValue.right.get shouldBe Query(predicates = List(Predicate("valid", OperationType.in)))
+    }
+
+    "return error with incorrect predicate fields" in {
+      (tpdo.areFieldsValid _).when("test", Set("invalid")).returns(Future.successful(false))
+
+      val query = ApiQuery(
+        fields = None,
+        predicates = Some(List(Predicate("invalid", OperationType.in))),
+        orderBy = None,
+        limit = None,
+        output = None,
+        aggregation = None
+      )
+
+      val result = query.validate("test", tpdo)
+
+      result.futureValue.left.get shouldBe List(InvalidPredicateField("invalid"))
+    }
+
+    "validate correct orderBy field" in {
+      (tpdo.areFieldsValid _).when("test", Set("valid")).returns(Future.successful(true))
+
+      val query = ApiQuery(
+        fields = None,
+        predicates = None,
+        orderBy = Some(List(QueryOrdering("valid", OrderDirection.asc))),
+        limit = None,
+        output = None,
+        aggregation = None
+      )
+
+      val result = query.validate("test", tpdo)
+
+      result.futureValue.right.get shouldBe Query(orderBy = List(QueryOrdering("valid", OrderDirection.asc)))
+    }
+
+    "return error with incorrect orderBy fields" in {
+      (tpdo.areFieldsValid _).when("test", Set("invalid")).returns(Future.successful(false))
+
+      val query = ApiQuery(
+        fields = None,
+        predicates = None,
+        orderBy = Some(List(QueryOrdering("invalid", OrderDirection.asc))),
+        limit = None,
+        output = None,
+        aggregation = None
+      )
+
+      val result = query.validate("test", tpdo)
+
+      result.futureValue.left.get shouldBe List(InvalidOrderByField("invalid"))
+    }
+
+    "validate correct aggregation field which exists in DB" in {
+      val attribute = Attribute(
+        name = "valid",
+        displayName = "valid",
+        dataType = DataType.Int,
+        cardinality = None,
+        keyType = KeyType.NonKey,
+        entity = "test"
+      )
+
+      (tpdo.areFieldsValid _).when("test", Set("valid")).returns(Future.successful(true))
+      (tpdo.getTableAttributesWithoutCounts _).when("test").returns(Future.successful(Some(List(attribute))))
+
+      val query = ApiQuery(
+        fields = Some(List("valid")),
+        predicates = None,
+        orderBy = None,
+        limit = None,
+        output = None,
+        aggregation = Some(Aggregation(field = "valid"))
+      )
+
+      val result = query.validate("test", tpdo)
+
+      result.futureValue.right.get shouldBe Query(fields = List("valid"), aggregation = Some(Aggregation(field = "valid")))
+    }
+
+    "return error with incorrect aggregation field which exists in DB" in {
```

Comment:
Can the test description make it apparent that the issue is that we're trying to aggregate on a non-aggregating field type?

Reply:
changed to be more descriptive